captum/attr/_core/neuron/neuron_ablation.py

```diff
@@ -0,0 +1,221 @@
+                            provided inputs, with each value providing the attribution
+                            of the corresponding input index.
+                            If a single tensor is provided as inputs, a single tensor is
+                            returned. If a tuple is provided for inputs, a tuple of
+                            corresponding sized tensors is returned.
+
+        Examples::
+
+            >>> # SimpleClassifier takes a single input tensor of size Nx4x4,
+            >>> # and returns an Nx3 tensor of class probabilities.
+            >>> # It contains an attribute conv1, which is an instance of nn.conv2d,
+            >>> # and the output of this layer has dimensions Nx12x3x3.
+            >>> net = SimpleClassifier()
+            >>> # Generating random input with size 2 x 4 x 4
+            >>> input = torch.randn(2, 4, 4)
+            >>> # Defining FeatureAblation interpreter
+            >>> ablator = FeatureAblation(net, net.conv1)
+            >>> # To compute neuron attribution, we need to provide the neuron
+            >>> # index for which attribution is desired. Since the layer output
+            >>> # is Nx12x3x3, we need a tuple in the form (0..11,0..2,0..2)
+            >>> # which indexes a particular neuron in the layer output.
+            >>> # For this example, we choose the index (4,1,2).
+            >>> # Computes neuron gradient for neuron with
+            >>> # index (4,1,2).
+            >>> # Computes ablation attribution, ablating each of the 16
+            >>> # scalar inputs independently.
+            >>> attr = ablator.attribute(input, neuron_index=(4,1,2))
+
+            >>> # Alternatively, we may want to ablate features in groups, e.g.
+            >>> # grouping each 2x2 square of the inputs and ablating them together.
+            >>> # This can be done by creating a feature mask as follows, which
+            >>> # defines the feature groups, e.g.:
+            >>> # +---+---+---+---+
+            >>> # | 0 | 0 | 1 | 1 |
+            >>> # +---+---+---+---+
+            >>> # | 0 | 0 | 1 | 1 |
+            >>> # +---+---+---+---+
+            >>> # | 2 | 2 | 3 | 3 |
+            >>> # +---+---+---+---+
+            >>> # | 2 | 2 | 3 | 3 |
+            >>> # +---+---+---+---+
+            >>> # With this mask, all inputs with the same value are ablated
+            >>> # simultaneously, and the attribution for each input in the same
+            >>> # group (0, 1, 2, and 3) per example are the same.
+            >>> # The attributions can be calculated as follows:
+            >>> # feature mask has dimensions 1 x 4 x 4
+            >>> feature_mask = torch.tensor([[[0,0,1,1],[0,0,1,1],
+            >>>                             [2,2,3,3],[2,2,3,3]]])
+            >>> attr = ablator.attribute(input, neuron_index=(4,1,2),
+            >>>                          feature_mask=feature_mask)
+        """
+
+        def neuron_forward_func(*args):
```

Comment:
Will additional `kwargs` also be supported here

Reply:
I don't think we need kwargs here, since inputs + additional_forward_args will always form a tuple, without keyword arguments?